upt_macports/tests/test_macports_backend.py

```diff
@@ -49,5 +49,19 @@ def test_port_error(self, mock_sub):
             self.macports_backend.package_versions('foo')
 
 
+class TestMacPortsCpanVersion(unittest.TestCase):
+    def setUp(self):
+        self.macports_backend = MacPortsBackend()
+        self.macports_backend.upt_pkg = upt.Package('foo', '42')
+        self.macports_backend.frontend = 'cpan'
+
+    def test_version_conversion(self):
+        converted = ['1', '1.2.3', '1.200.0', '1.200.0', '1.20.0']
+        upstream = ['1', '1.2.3', '1.2', '1.20', '1.02']
```

Comment:
thank you for adding tests! In the conversion code (both in MacPorts and here), a possible leading ```v``` in the upstream version is stripped in the function before doing the conversion. Nevertheless, you will be sometimes passing in an upstream version with a ```v``` present and you probably want to check that the conversion works correctly in such cases as well